File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            map.put(nums[i],1+map.getOrDefault(nums[i],0));
             if(map.get(nums[i])>n/2)
                return nums[i];
        }
        
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count, candidate = 0, 0
        for i in range(len(nums)):
            if count == 0:
                candidate, count = nums[i], 1
            elif candidate == nums[i]:
                count += 1
            else:
                count -= 1
        return candidate